File: solutions/cpp14/easy/rectangle-area/solution-approach1.cpp
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/challenges/rectangle-area/problem?isFullScreen=true
// Problem     Rectangle Area
// Difficulty  Easy
// Subdomain   Inheritance
// Platform    HackerRank
// Language    cpp14
// Status      Accepted
// Submitted   2026-08-13, 05:14 p.m.
// ──────────────────────────────────────────────────


/*
 * Create classes Rectangle and RectangleArea
 */
 class Rectangle {
protected:
    int width;
    int height;

public:
    // Display method to print width and height
    virtual void display() const {
        cout << width << " " << height << endl;
    }
};
class RectangleArea : public Rectangle {
public:
    // Reads width and height from standard input
    void read_input() {
        cin >> width >> height;
    }

    // Overloaded display method to print the calculated area
    void display() const override {
        cout << width * height << endl;
    }
};

File: solutions/python3/easy/list-comprehensions/solution-approach1.py
# ──────────────────────────────────────────────────
# Link        https://www.hackerrank.com/challenges/list-comprehensions/problem?isFullScreen=true
# Problem     List Comprehensions
# Difficulty  Easy
# Subdomain   Basic Data Types
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-09-15, 11:48 p.m.
# ──────────────────────────────────────────────────

if __name__ == '__main__':
    x = int(input())
    y = int(input())
    z = int(input())
    n = int(input())
    
    result = [[i, j, k] for i in range(x + 1) for j in range(y + 1) for k in range(z + 1) if (i + j + k) != n]
    print(result)
